changelog/3691.bugfix.rst

```diff
@@ -0,0 +1,2 @@
+Python 2: safely format warning message about passing unicode strings to ``warnings.warn``, which may cause
+surprising ``MemoryError`` exception when monkey patching ``warnings.warn`` itself.
```

Comment:
*(comment for my own education about pytest's documentation)*

In Hypothesis we have the [`intersphinx_mapping`](https://github.com/HypothesisWorks/hypothesis/blob/192716ef679864e8685a6d759e609d70edfb25b7/hypothesis-python/docs/conf.py#L68-L75) configured so that we can write these as:

    surprising :class:`python:MemoryError` exception when monkey 
    patching :func:`python:warnings.warn` itself.

It allows for somewhat nicer formatting, but more importantly adds a hyperlink - and can check that the link target exists, which caught dozens of typos in our docs.

Has pytest decided not to do this, or just not considered it?  It can be fiddly at times but also pretty useful.

Reply:
i don't recall use taking a deeper look into this so it may be worth trying it - ideally someone with experience setting it up

Reply:
@Zac-HD thanks a lot for the pointer, actually or `intersphinx_mapping` is already configured. I tested locally your changes and they actually do work. 😁 

We should remember to use those sphinx references more often. 👍

Reply:
Hmm actually I spoke too soon. While it generates the proper docs, our linter complains:

```
ERROR changelog/3691.bugfix.rst:1 Unknown interpreted text role "func".
ERROR changelog/3691.bugfix.rst:1 Unknown interpreted text role "class".
ERROR changelog/3691.bugfix.rst:1 Unknown interpreted text role "func".
ERROR changelog/4192.bugfix.rst:1 Unknown interpreted text role "func".
ERROR changelog/4192.bugfix.rst:1 Unknown interpreted text role "func".
```